src/H3MP/Utils/WristMenuButtons.cs

```diff
@@ -0,0 +1,280 @@
+			for (var i = 0; i < buttons.Count; ++i)
+			{
+				var button = buttons[i];
+				WristMenu.Buttons.Add(button);
+
+				var image = button.GetComponent<Image>();
+				var pointable = button.GetComponent<FVRWristMenuPointableButton>();
+				var index = originalButtonsLength + i;
+
+				pointable.ButtonIndex = index;
+				images[index] = image;
+			}			
+		}
+
+		private IEnumerable<Button> CreateButtons(Transform canvas)
+		{
+			var source = canvas.Find("Button_3_ReloadScene").gameObject;
+			Button CreateButton(string posName, string name, string text, Func<Button, UnityAction> callbackFactory)
+			{
+				// Move down source button
+				var posTransform = canvas.Find(posName);
+				posTransform.position += BUTTON_PLACEMENT_DIFF * Vector3.down;
+
+				// Create our button
+				var dest = GameObject.Instantiate(source, posTransform.position, posTransform.rotation, posTransform.parent);
+				dest.transform.position += BUTTON_PLACEMENT_DIFF * Vector3.up;
+				dest.name = name;
+				dest.GetComponentInChildren<Text>().text = text;
+
+				// Destroy base button to remove persistent onClick listener & add our own
+				var button = dest.GetComponent<Button>();
+				Component.DestroyImmediate(button);			
+				button = dest.AddComponent<Button>();
+
+				button.onClick.AddListener(callbackFactory(button));
+
+				// Shift all base buttons below down
+				var btnIndex = _wristMenuBaseButtons.IndexOf(posName);
+				for (var i = btnIndex + 1; i < _wristMenuBaseButtons.Count; i++)
+				{
+					MoveButton(_wristMenuBaseButtons[i]);
+				}
+
+				return button;
+			}
+
+			void MoveButton(string posName)
+			{
+				var posTransform = canvas.Find(posName);
+				posTransform.position += BUTTON_PLACEMENT_DIFF * Vector3.down;
+			}
+		
+			// Add new buttons from top to bottom
+			yield return CreateButton("Button_1_OptionsPanel", "H3MP_Panel", "Spawn H3MP Panel", SpawnPanel);
+			yield return CreateButton("Button_3_ReloadScene", "H3MP_Privacy", PrivacyManager.PrivacyText, PrivacySelector);
+			yield return CreateButton("Button_9_BackToMainMenu", "H3MP_Disconnect", DISCONNECT_TEXT, LeaveLobby);
+		}
+
+		// H3MP_Panel onClick
+		private UnityAction SpawnPanel(Button button)
+		{
+			return () =>
+			{
+				WristMenu.Aud.PlayOneShot(WristMenu.AudClip_Engage, 1f);
+
+				FVRViveHand hand = GM.CurrentPlayerBody.RightHand.transform.Find("PointingLaser").gameObject.activeInHierarchy
+					? GM.CurrentPlayerBody.LeftHand.GetComponent<FVRViveHand>()
+					: GM.CurrentPlayerBody.RightHand.GetComponent<FVRViveHand>();
+
+				hand.RetrieveObject(GetPanel());
+				_log.LogDebug("Spawned panel");
+			};
+		}
+
+		// H3MP_Privacy onClick
+		private UnityAction PrivacySelector(Button button)
+		{
+			return () =>
+			{
+				WristMenu.Aud.PlayOneShot(WristMenu.AudClip_Engage, 1f);
+
+				PrivacyManager.CyclePrivacy();
+
+				var buttonText = button.GetComponentInChildren<Text>().text = PrivacyManager.PrivacyText;
```

Comment:
Cache this component by getting it outside the lambda, and capturing the variable.